### PR TITLE
fix(status): rudimentary support for filenames with spaces

### PR DESCRIPTION
**Pull request by claytonrcarter**
The regex in `StatusEntry::try_from()` was mistaken parsing status lines for files with spaces in such a way to discard the first word in the path name. This attempts to make that less likely by making the regex more specific, but I don't think it's really unavoidable without completely reworking this part of the code to be less flexible. eg if we parsed each of the 1/2/u statuses separately, then we could be more specific about how many fields we need to match/capture for each option.

I suspect that this issue was already rare, and that this fix will make it even moreso. Given that, I feel like a more robust/complex fix probably isn't warranted. With this fix, files like "r101 project.txt" or "Coffee notes.txt" will work, but "R101 project.txt" or "coffee notes.txt" ("R101" looks like a "change score", and "coffee" is a ~~valid~~ hex value.) (**Edit:** 🤦 `coffee` is not a "hex word"; `c0ffee` is, though.)

The options that come to mind to tighten this up further:
- parse each 1/2/u status separately, making field counts and patterns more discrete
- require a minimum number of chars in an object ID (on my Mac, the longest "hex word" in /usr/share/dict/words is 7 chars, so requiring at least 8 chars in an OID would eliminate "coffee"-style false positives)

### CODE REVIEW

File: git-branchless/tests/test_amend.rs
Comment (claytonrcarter) [inline git-branchless/tests/test_amend.rs:426]: I added a test for `git amend` because that's where I first encountered this, while working on some documentation.

File: git-branchless/tests/test_amend.rs
Comment (claytonrcarter) [inline git-branchless/tests/test_amend.rs:421]: Oh, this may be leftover from the test I copied to create this one. 🤔

File: git-branchless-lib/src/git/status.rs
Comment (claytonrcarter) [inline git-branchless-lib/src/git/status.rs:204]: There are numerous ways to update this pattern that could make this less likely to happen, but I don't think we can really eliminate it so long as we use 1 regex for all these porcelain status types. I go into some of them in the PR/commit description.

In particular, the git docs at https://git-scm.com/docs/git-status#_porcelain_format_version_2 state that the `-z` flag (which is used here) causes path names not to be quoted, which ends up leaving the path name a little ambiguous for us.

File: git-branchless-lib/tests/test_status.rs
Comment (claytonrcarter) [inline git-branchless-lib/tests/test_status.rs:25]: These are all identical to the preceding assertions, except for the file name.

File: git-branchless/tests/test_amend.rs
Comment (arxanas) [inline git-branchless/tests/test_amend.rs:431]:
```suggestion
```

File: git-branchless-lib/src/git/status.rs
Comment (arxanas) [reply git-branchless-lib/src/git/status.rs:204]: Oops, this is unfortunate. Yeah, refactoring it further sounds non-trivial, so we can hack around it for now.

File: git-branchless/tests/test_amend.rs
Comment (arxanas) [reply git-branchless/tests/test_amend.rs:421]: I'm assuming this is referring to the commented-out lines below (and that this comment is on the wrong lines now due to a force-push?).

**Review by arxanas — APPROVED** [review]
Thanks!

File: git-branchless/tests/test_amend.rs
Comment (claytonrcarter) [reply git-branchless/tests/test_amend.rs:421]: No, this is the lines I was intending, but I also think that I was wrong here. I was thinking that I wasn't doing anything w/ committer/author data, but then realized that this is probably intended to ensure that the commit IDs generated by `amend` are deterministic for the tests.

File: git-branchless/tests/test_amend.rs
Comment (claytonrcarter) [reply git-branchless/tests/test_amend.rs]: Oops! I left this in from when I was debugging this. Thanks for catching it!

File: git-branchless-lib/src/git/status.rs
Comment (claytonrcarter) [reply git-branchless-lib/src/git/status.rs]: Thanks. NOTE that I updated this line before merge because I had mistakenly doubled the square brackets. This is the change I made just now (`[[a-f]\d]+` → `[a-f\d]+`:
```diff
-                r#"([[a-f]\d]+ ){2,3}([CR]\d{1,3} )?"#,
+                r#"([a-f\d]+ ){2,3}([CR]\d{1,3} )?"#,

**Pull request merged**
